### PR TITLE
SystemUtils with tests

### PR DESCRIPTION
**Pull request by IhateTrains**
Stuff like GetAllFilesInFolder, TryCreateFolder etc.

Tests are invented by me, C++ version seems to be missing them.

### CODE REVIEW

File: commonItems/SystemUtils.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+            }
+            return new SortedSet<string>();
+        }
+
+        public static bool TryCreateFolder(string path) {
+            try {
+                return Directory.CreateDirectory(path).Exists;
+            } catch (Exception e) {
+                Logger.Log(LogLevel.Error, "Could not create directory: " + path + " : " + e);
+                return false;
```
Comment (Idhrendur) [inline commonItems/SystemUtils.cs:32]: Isn't it more idiomatic in C# to rethrow the exception instead of converting to a success/fail return type?

Comment (Idhrendur) [reply commonItems/SystemUtils.cs:32]: Same for the other functions, possibly even functions in past PRs

File: commonItems/SystemUtils.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            try {
+                Directory.Move(sourceFolder, destFolder);
+                return true;
+            } catch (Exception e) {
+                Logger.Log(LogLevel.Error, "Could not rename folder: " + e);
+                return false;
+            }
+        }
+
+        public static bool DoesFileExist(string path) {
```
Comment (Idhrendur) [inline commonItems/SystemUtils.cs:93]: Given how much easier this is in C#, you might drop this utility function entirely.

File: commonItems/SystemUtils.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            return File.Exists(path);
+        }
+
+        public static bool DoesFolderExist(string path) {
```
Comment (Idhrendur) [inline commonItems/SystemUtils.cs:97]: Given how much easier this is in C#, you might drop this utility function entirely.

File: commonItems/UnitTests/SystemUtilsTests.cs [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+using Xunit;
+
+namespace commonItems.UnitTests {
+    [Collection("Sequential")]
+    [CollectionDefinition("Sequential", DisableParallelization = true)]
+    public class SystemUtilsTests {
+        readonly string testFilesPath = "UnitTests/TestFiles";
+        [Fact] public void GetAllFilesInFolderDoesntWorkRecursively() {
+            var files = SystemUtils.GetAllFilesInFolder(testFilesPath);
+            Assert.Single(files);
```
Comment (Idhrendur) [inline commonItems/UnitTests/SystemUtilsTests.cs:12]: Can you make it more explicit? It'd be nice to have it show the format of the files in the SortedSet. For instance, is the path included in the filename or not? If I were using this library, I could verify that be looking at the test.

File: commonItems/UnitTests/SystemUtilsTests.cs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        }
+        [Fact]
+        public void GetAllFilesInFolderReturnsEmptySetOnNonexistentFolder() {
+            var files = SystemUtils.GetAllFilesInFolder(testFilesPath + "/missingDir");
+            Assert.Empty(files);
+        }
+        [Fact]
+        public void GetAllFilesInFolderRecursiveWorkRecursively() {
+            var files = SystemUtils.GetAllFilesInFolderRecursive(testFilesPath);
+            Assert.Equal(4, files.Count);
```
Comment (Idhrendur) [inline commonItems/UnitTests/SystemUtilsTests.cs:22]: Ditto

File: commonItems/UnitTests/SystemUtilsTests.cs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        }
+        [Fact]
+        public void GetAllFilesInFolderRecursiveReturnsEmptySetOnNonexistentFolder() {
+            var files = SystemUtils.GetAllFilesInFolderRecursive(testFilesPath + "/missingDir");
+            Assert.Empty(files);
+        }
+        [Fact]
+        public void GetAllSubfoldersGetsSubfolders() {
+            var subfolders = SystemUtils.GetAllSubfolders(testFilesPath);
+            Assert.Equal(2, subfolders.Count);
```
Comment (Idhrendur) [inline commonItems/UnitTests/SystemUtilsTests.cs:32]: Ditto

**Review by Idhrendur — APPROVED** [review]

File: commonItems/SystemUtils.cs (hunk 1 above)
Comment (IhateTrains) [reply commonItems/SystemUtils.cs:32]: Returning success is also used, so I will use it for now to stay close to C++ version. Rethrowing in these functions would be useful if we wanted to have advanced handling in the converter for specific exceptions, but does any converter do that?
I've changed the names to `TryX` for consistency, though.

**Pull request merged**